test/Common.test.cpp

```diff
@@ -56,6 +56,10 @@ TEST(String, splitOnUnquoted)
 	EXPECT_EQ((NAS2D::StringList{"", "abc", "def"}), NAS2D::splitOnUnquoted("\"\".abc.def", '.'));
 	EXPECT_EQ((NAS2D::StringList{"abc", "", "def"}), NAS2D::splitOnUnquoted("abc.\"\".def", '.'));
 	EXPECT_EQ((NAS2D::StringList{"abc", "def", ""}), NAS2D::splitOnUnquoted("abc.def.\"\"", '.'));
+	EXPECT_EQ((NAS2D::StringList{"Hello", "Hello World", "World"}), NAS2D::splitOnUnquoted("Hello \"Hello World\" World", ' '));
+	EXPECT_EQ((NAS2D::StringList{"Hello", " ", "World"}), NAS2D::splitOnUnquoted("Hello \" \" World", ' '));
+	EXPECT_EQ((NAS2D::StringList{"Hello", "", "World"}), NAS2D::splitOnUnquoted("Hello \"\" World", ' '));
```

Comment:
What about quotes in the middle of a string? I assume those are ignored?
```
EXPECT_EQ((NAS2D::StringList{"abc"}), NAS2D::splitOnUnquoted("a\"b\"c")); // Or broken into {"a", "b", "c"}?
```

----

I suppose you've actually tested this, though it wasn't immediately obvious. The parser is stateful. Does it handle arbitrary mixes of quoted and unquoted strings? A specific case I don't see is two quoted strings separated by an unquoted string.
```
EXPECT_EQ((NAS2D::StringList{"a", "b", "c"}), NAS2D::splitOnUnquoted("\"a\",b,\"c\""));
```

Reply:
It's a simple on/off state like block quotes in C++ `/*...*/` attempting to nest either will make you sad.

I wrote this one specifically for a `KeyValueParser`.

I wrote one for an `ArgumentParser` that uses a different algorithm. It uses a stream and checks the first character of an extracted word to determine if it's in a quote and checks the last character of an extracted word to determine when it's not anymore. I did it that way because the argument parser does not preserve the quotes like the key value parser does.

Reply:
A simple on/off switch is a simple and sensible way to handle things.

From the test cases I read through, it looked like quotes were stripped. I suspect it would be more useful to strip the quotes than to preserve them. I'm wondering if I misunderstood what you're saying about preserving quotes. Or perhaps that was a later change I didn't check in detail.